node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -144,19 +144,16 @@ impl<C: Context> ActiveValidator<C> {
             return vec![Effect::WeEquivocated(evidence.clone())];
         }
         let mut effects = vec![];
-        if self.should_send_confirmation(vhash, now, state) {
-            let panorama = match state.confirmation_panorama(self.vidx, vhash) {
-                None => return effects,
-                Some(panorama) => panorama,
-            };
+        if self.should_send_confirmation(uhash, now, state) {
+            let panorama = state.confirmation_panorama(self.vidx, uhash);
             if panorama.has_correct() {
                 let confirmation_unit = self.new_unit(panorama, now, None, state, instance_id, rng);
                 let vv = ValidVertex(Vertex::Unit(confirmation_unit));
-                effects.push(Effect::NewVertex(vv));
+                effects.push(Effect::NewVertex(vv))
```

Comment:
Why remove the semicolon? This has `()` type anyway, does it?

Reply:
I don't remember doing it. Maybe it was clippy or cargo fmt. Build should catch it if that's an error.